mantidproject/mantid
# Muon Phasequad Table

**Description of work.**
In the phase tab of Muon Analysis:
Removed ui file and replaced with code.
Added phasetable selector
Added table of phasequads

**To test:**
open Muon Analysis and load MUSR62260 
(it may help here to remove all the groups from the analysis in the grouping tab)
In the phase tab, try add a phasequad by pressing the plus at the bottom of the table (should not be allowed)
Now create a phasetable
Add a phasequad
Untick and tick analyse for both real and imaginary parts
Add a few more
Try pressing the - button to remove the last row, or select some rows and click the - button to remove selected rows
With some existing phasequads, create a new different phasetable and then use the phase table selector to change table (the phasequads should update)
try other stuff with groups and pairs to make sure nothing is broken there

Fixes #29611 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
